Utilities/sync.h

```diff
@@ -76,6 +78,53 @@ inline int futex(volatile void* uaddr, int futex_op, uint val, const timespec* t
 {
 #ifdef __linux__
 	return syscall(SYS_futex, uaddr, futex_op, static_cast<int>(val), timeout, nullptr, static_cast<int>(mask));
+#elif __APPLE__
+	switch (futex_op)
+	{
+	case FUTEX_WAIT_PRIVATE:
+	case FUTEX_WAIT_BITSET_PRIVATE:
+	{
+		for (;;) 
+		{
+	      long ret = 0;
+	      if (timeout)
+		  {
+			const uint64_t nsec = timeout->tv_nsec + timeout->tv_sec * 1000000000ull;
+	        ret = os_sync_wait_on_address_with_timeout(const_cast<void*>(uaddr), static_cast<uint64_t>(val), sizeof(uint), OS_SYNC_WAIT_ON_ADDRESS_NONE, OS_CLOCK_MACH_ABSOLUTE_TIME, nsec);
+	      }
+		  else
+		  {
+	        ret = os_sync_wait_on_address(const_cast<void*>(uaddr), static_cast<uint64_t>(val), sizeof(uint), OS_SYNC_WAIT_ON_ADDRESS_NONE);
+	      }
+	      if ((ret < 0) && (errno == ETIMEDOUT))
+		  {
+	        return -ETIMEDOUT;
+		  }
+			
+	      if ((ret < 0) && ((errno == EINTR) || (errno == EFAULT))) 
```

Comment:
If there is an an issue it would be caught in atomic.cpp. If re-iteration is required atomic.cpp would also take care of that.
So I think this loop and check can be removed.